tests/wakunode_rest/test_rest_store.nim

```diff
@@ -534,3 +534,67 @@ procSuite "Waku v2 Rest API - Store":
     await restServer.stop()
     await restServer.closeWait()
     await node.stop()
+
+
+  asyncTest "retrieve historical messages from a self-store-node":
+    ## This test aims to validate the correct message retrieval for a store-node which exposes
+    ## a REST server.
+
+    # Given
+    let node = testWakuNode()
+    await node.start()
+
+    let restPort = Port(58014)
```

Comment:
```suggestion
    let restPort = Port(0)
```

Reply:
Thanks for the suggestion! However, I won't apply it as it resulted in an exception:
```
    Unhandled error: Communication failed while sending/receiving request, http error [HttpConnectionError]: Could not connect to remote host [RestCommunicationError]

```